### PR TITLE
Chain periodic sync completion with CustomSyncWorker  

### PR DESCRIPTION
**Pull request by Lentumunai-Mark**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes [link to issue]

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by dubdabasoduba — APPROVED** [review]

File: android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncBroadcaster.kt [hunk 1]
```diff
@@ -103,12 +103,26 @@ constructor(
...
-            } else it.currentSyncJobStatus,
+            if (status.lastSyncJobStatus != null) {
+              CurrentSyncJobStatus.Succeeded(
+                (status.lastSyncJobStatus as LastSyncJobStatus).timestamp,
+              )
+            } else status.currentSyncJobStatus,
+          )
+        }
+
+        if (status.currentSyncJobStatus is CurrentSyncJobStatus.Succeeded) {
```
Comment (ndegwamartin) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncBroadcaster.kt:118]: I think we should add an **OR** condition here with `status.lastSyncJobStatus != null` as well since it represents a `CurrentSyncJobStatus.Succeeded` status as well _(hence code block L110-L113)_

Comment (ndegwamartin) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncBroadcaster.kt:118]: Essentially we should be looking for two things, that is is not null and it Succeeded _(LastSyncJobStatus.Succeeded)_

Check out the Sealed class here - https://github.com/google/android-fhir/blob/507e89cc8b75c7b6667c8bff47b914c3ac94a770/engine/src/main/java/com/google/android/fhir/sync/SyncJobStatus.kt#L46

File: android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncBroadcaster.kt [hunk 2]
```diff
@@ -103,12 +103,26 @@ constructor(
...
   ) {
-    this.onEach {
+    this.onEach { status ->
         syncListenerManager.onSyncListeners.forEach { onSyncListener ->
+          Timber.d("fhir sync worker...")
           onSyncListener.onSync(
-            if (it.lastSyncJobStatus != null) {
-              CurrentSyncJobStatus.Succeeded((it.lastSyncJobStatus as LastSyncJobStatus).timestamp)
-            } else it.currentSyncJobStatus,
+            if (status.lastSyncJobStatus != null) {
```
Comment (ndegwamartin) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/sync/SyncBroadcaster.kt:110]: Lets also update this existing check with the same condition. i.e.
```suggestion
            if (status.lastSyncJobStatus != null && status.lastSyncJobStatus is LastSyncJobStatus.Succeeded) {
```

**Comment by ellykits** [comment]
@Lentumunai-Mark What happens when the custom sync is complete? Are you observing any states?

**Comment by Lentumunai-Mark** [comment]
closing this in favor of #3650
